concent_api/api-integration-force-get-task-result-test.py

```diff
@@ -1,197 +1,362 @@
+                    verify  = False,
+                  )
+    return response
 
-def get_force_get_task_result(task_id, current_time, size, checksum):
 
-    compute_task_def = message.ComputeTaskDef()
-    compute_task_def['task_id'] = task_id
-    compute_task_def['deadline'] = current_time + 60
-    task_to_compute = message.TaskToCompute(
-        compute_task_def = compute_task_def
-    )
+def get_force_get_task_result(task_id, current_time, size, checksum, task_deadline_offset=60):
+    task_to_compute      = create_task_to_compute(current_time, task_id, task_deadline_offset)
+
     report_computed_task = message.ReportComputedTask(
-        task_to_compute = task_to_compute,
-        size            = size,
-        checksum        = checksum,
+        task_to_compute  = task_to_compute,
+        size             = size,
+        checksum         = checksum,
     )
 
-    with freeze_time(timestamp_to_isoformat(current_time)):
-        force_get_task_result = message.concents.ForceGetTaskResult(
-            report_computed_task = report_computed_task,
-        )
+    force_get_task_result = message.concents.ForceGetTaskResult(
+        report_computed_task = report_computed_task,
+    )
 
     return force_get_task_result
 
 
-def main():
-    cluster_url     = parse_command_line(sys.argv)
-    current_time    = get_current_utc_timestamp()
-    task_id         = str(random.randrange(1, 100000))
-
-    (response_status_code, file_size, file_check_sum) = upload_new_file_on_cluster(
-        task_id = task_id,
-        part_id = '0',
-        current_time = current_time,
-    )
-    if response_status_code == 200:
-        print('\nCreated file with task_id {}. Checksum of this file is {}, and size of this file is {}.\n'.format(task_id, file_check_sum, file_size))
-    else:
-        print('File has not been stored on cluster')
+class count_fails(object):
+    """
+    Decorator that wraps a test functions for intercepting assertions and counting them.
+    """
+    instances = []
+
+    def __init__(self, fun):
+        self._fun     = fun
+        self.__name__ = fun.__name__
+        self.failed  = False
+        count_fails.instances.append(self)
+
+    def __call__(self, *args, **kwargs):
+        try:
+            print("Running TC: " + self.__name__)
+            return self._fun(*args, **kwargs)
+        except AssertionError as e:
+            print("{}: FAILED".format(self.__name__))
+            print(e)
+            self.failed = True
+
+    @classmethod
+    def get_fails(cls):
+        return sum([instance.failed for instance in cls.instances])
+
+
+@count_fails
+def case_4d_concent_notifies_the_provider_that_task_results_are_ready(cluster_url, current_time, task_id):
+    file_check_sum, file_content, file_size, headers = prepare_file_for_transfer('0', task_id, TOKEN_EXPIRATION_TIME)
+
+    response = upload_file(file_content, headers)
+
+    assert response.status_code == 200, 'File has not been stored on cluster'
```

Comment:
Please do not use `assert` for checking expectations. We use `assert` for checking assumptions about the code. If we do `assert X` then `X` is expected to be universally true unless the code has a bug. This is not the case here. This may be false when server returns an unexpected response.

Reply:
done